src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,332 @@
+    rootContainerInstance : Container,
+  ) : boolean {
+    // console.log('finalizeInitialChildren');
+    // setInitialProperties(testElement, type, props, rootContainerInstance);
+    return false;
+  },
+
+  prepareUpdate(
+    testElement : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    hostContext : Object,
+  ) : boolean {
+    return true;
+  },
+
+  commitUpdate(
+    instance : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    instance.update(type, newProps);
+  },
+
+  commitMount(
+    instance : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Object,
+    internalInstanceHandle : Object
+  ) : void {
+    // Noop
+  },
+
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number'
+    );
+  },
+
+  resetTextContent(testElement : Instance) : void {},
+
+  createTextInstance(
+    text : string,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object
+  ) : TextInstance {
+    var inst = {
+      text : text,
+      id: instanceCounter++,
+      rootContainerInstance,
+      toJSON: () => isNaN(+inst.text) ? inst.text : +inst.text,
+    };
+    // Hide from unit tests
+    Object.defineProperty(inst, 'id', { value: inst.id, enumerable: false });
+    return inst;
+  },
+
+  commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string) : void {
+    textInstance.text = newText;
+  },
+
+  appendChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    parentInstance.insertBefore(beforeChild, child);
+  },
+
+  removeChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.removeChild(child);
+  },
+
+  scheduleAnimationCallback: window.requestAnimationFrame,
+
+  scheduleDeferredCallback: window.requestIdleCallback,
+
+  useSyncScheduling: true,
+
+  getPublicInstance(ref) {
+    const createNodeMock = ref.rootContainerInstance.createNodeMock;
+    return createNodeMock(ref);
+  },
+
+});
+
+var defaultTestOptions = {
+  createNodeMock: function() {
+    return null;
+  },
+};
+
+var ReactTestFiberRenderer = {
+  create(element: ReactElement<any>, options: TestRendererOptions) {
+    var createNodeMock = defaultTestOptions.createNodeMock;
+    if (options && typeof options.createNodeMock === 'function') {
+      createNodeMock = options.createNodeMock;
+    }
+    var container = new TestContainer('<default>', createNodeMock);
+    var root = TestRenderer.createContainer(container);
+    if (root) {
```

Comment:
You'd need to see why other renderers didn't need this. It probably inferred that `OpaqueNode` type can be null, maybe from some other method definition.